### ISSUE TITLE
Search command inside a module fails to see declarations coming by module inclusion

### ISSUE THREAD

**Issue by pi8027** [coq/coq#12525]
#### Description of the problem

```coq
Require Import Arith.

Search Nat.lcm. (* prints a bunch of lemmas *)

Module M.

Search Nat.lcm. (* prints nothing *)
```

But if we search for `Nat.add` instead, it works. I guess this is a bug of the new `Search` functionality introduced in 8.12 (#8855).

#### Coq Version

master

**Comment by Zimmi48** [comment]
Ouch, that's a serious regression!

**Comment by Zimmi48** [comment]
cc @herbelin

**Comment by herbelin** [comment]
A priori, this is not related to the new `Search` but to suspected changes in the semantics of `Include`. Here is a reduced example in the empty state:
```
Module A. Axiom a:Prop. Axiom b:a. End A.     
Module B. Include A. End B.
Search B.a.
(*
B.b: B.a
A.b: A.a
*)
Module M.
Search B.a.
(*
A.b: A.a
*)
```

**Comment by herbelin** [comment]
Or maybe in the equality of kernel names across `Include`??

**Comment by herbelin** [comment]
Update: This is related to the new `Search` which relies on the table of kinds (`csttab` currently in `dumpglob.ml`) which seems to have a problem with `Include`. I'm investigating.

**Comment by herbelin** [comment]
The bug comes from an incorrect use of the delta-resolver when in a `Module`. The call to `Module M` reinitializes the delta resolver and this breaks the delta-resolution of names brought by the `Include`. Therefore `B.a` (in the short example) is not anymore resolved into its canonical name. See #12537 for a fix.

**Comment by ejgallego** [comment]
Thanks to @herbelin we understood a bit more about this bug.

Definitions are skipped because of new filtering added regarding the kind of the constant w.r.t. the table maintained in `Dumpglob`.  The table gets the incorrect names due to the wrong delta resolver.

A more conservative fix towards 8.12 would be to simply disable the kind check for now.
